src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -54,21 +73,37 @@
     List<Object> unmergedFields = new ArrayList<>();
     for (String fieldSpec : fieldSpecs) {
       Object topLevel = InitValueConfig.create();
-      if (initValueConfigMap.containsKey(fieldSpec)) {
+      String[] equalsParts = fieldSpec.split("[=]");
+      if (equalsParts.length > 2) {
+        throw new IllegalArgumentException("Inconsistent: found multiple '=' characters");
+      }
+      if (equalsParts.length == 2) {
+        topLevel = InitValueConfig.createWithValue(equalsParts[1]);
+      } else if (initValueConfigMap.containsKey(fieldSpec)) {
         topLevel = initValueConfigMap.get(fieldSpec);
       }
-      String[] stringParts = fieldSpec.split("[.]");
-      for (int i = stringParts.length - 1; i >= 0; i--) {
-        Matcher listMatcher = fieldListPattern.matcher(stringParts[i]);
-        if (listMatcher.matches()) {
-          String name = listMatcher.group(1);
+
+      String toMatch = equalsParts[0];
+      while (toMatch != null) {
+        Matcher structureMatcher = fieldStructurePattern.matcher(toMatch);
+        Matcher listMatcher = fieldListPattern.matcher(toMatch);
+        Matcher mapMatcher = fieldMapPattern.matcher(toMatch);
+        if (structureMatcher.matches()) {
+          String key = structureMatcher.group(2);
+          topLevel = new AutoValue_FieldStructureParser_FieldSpec(key, topLevel);
```

Comment:
I would prefer this particular part of the code to not know that FieldSpec uses AutoValue; could you make create methods in the spec classes to encapsulate that?